File: ACO/v3/Human.py
import pandas


class Human:
    __slots__ = ('id', 'name', 'ability_score')

    def __init__(self, id: float, name: str, abilityScore: float) -> None:
        self.id = int(id)
        self.name = name
        self.ability_score = abilityScore


def get_list_human(filename: str = 'HumanAbilityScore.csv') -> list:
    df = pandas.read_csv(filename)
    return [Human(row[1]['human_id'], row[1]['name'], row[1]['abilityScore'])
            for row in df.iterrows()]

File: ACO/Human.js
import {DataFrame} from 'dataframe-js'
import { readFileSync } from 'fs';

export class Human {
    constructor(id, name, abilityScore) {
        this.id = parseInt(id);
        this.name = name;
        this.ability_score = parseFloat(abilityScore);
    }
}

export const getListHuman = (filename = 'HumanAbilityScore.csv') => {
    const csvData = readFileSync(filename, 'utf8');

    const csvArray = csvData
        .trim()
        .split('\n')
        .map(line => line.split(','))
        .map(cols => ({
            human_id: cols[0],
            name: cols[1],
            abilityScore: cols[2],
    }));
    csvArray.shift()
    const listHuman = csvArray.map(row => new Human(row.human_id, row.name, row.abilityScore));
    return listHuman
}

